build.cake

```diff
@@ -42,7 +43,63 @@ var password = EnvironmentVariable("CAKE_GITHUB_PASSWORD");
 
 Setup(() =>
 {
-    Information("Building version {0} of Cake.", semVersion);
+    if (isRunningOnWindows)
+    {
+        Information("Calculating Semantic Version");
+        if (AppVeyor.IsRunningOnAppVeyor)
+        {
+            GitVersion(new GitVersionSettings {
+                UpdateAssemblyInfoFilePath = "./src/SolutionInfo.cs",
+                UpdateAssemblyInfo = true,
+                OutputType = GitVersionOutput.BuildServer
+            });
+
+            version = EnvironmentVariable("GitVersion_MajorMinorPatch");
+            semVersion = EnvironmentVariable("GitVersion_LegacySemVerPadded");
+            milestone = string.Concat("v", version);
+
+            // Due to the way that GitVersion is executed on AppVeyor, the Environment Variables although populated
+            // are not accessible yet, so have to run GitVersion again, using OutputType of JSON
+            if(string.IsNullOrEmpty(semVersion))
```

Comment:
The contents of this if statement is essentially duplicated in the else statement below.  So I think we could remove this duplication, and just always run this code, just make sure to run the above code, if running on AppVeyor.